src/linker/Linker.Dataflow/ReflectionPatternContext.cs

```diff
@@ -95,6 +96,19 @@ public void RecordUnrecognizedPattern (int messageCode, string message)
 				_context.ReflectionPatternRecorder.UnrecognizedReflectionAccessPattern (Origin, Source, Instruction, MemberWithRequirements, message, messageCode);
 		}
 
+		public void RecordUnrecognizedPattern (DiagnosticId messageCode, string message)
+		{
+#if DEBUG
+			if (!_patternAnalysisAttempted)
+				throw new InvalidOperationException ($"Internal error: To correctly report all patterns, when starting to analyze a pattern the AnalyzingPattern must be called first. {Source} -> {MemberWithRequirements}");
+
+			_patternReported = true;
+#endif
+
+			if (ReportingEnabled)
+				_context.ReflectionPatternRecorder.UnrecognizedReflectionAccessPattern (Origin, Source, Instruction, MemberWithRequirements, message, (int) messageCode);
```

Comment:
Why not just call the other overload by casting the messageCode? This feels like unnecessary duplication.